File: mobile/lib/models/guest.dart
class Guest {
  final String id;
  final String tenantId;
  final String name;
  final String? email;
  final String? phone;
  final String rsvpStatus;
  final bool plusOne;
  final String? dietaryRestrictions;
  final String? tableId;
  final String? side;
  final String? notes;

  const Guest({
    required this.id,
    required this.tenantId,
    required this.name,
    this.email,
    this.phone,
    this.rsvpStatus = 'pending',
    this.plusOne = false,
    this.dietaryRestrictions,
    this.tableId,
    this.side,
    this.notes,
  });

  factory Guest.fromJson(Map<String, dynamic> json) {
    return Guest(
      id: json['id'] as String,
      tenantId: json['tenantId'] as String,
      name: json['name'] as String,
      email: json['email'] as String?,
      phone: json['phone'] as String?,
      rsvpStatus: json['rsvpStatus'] as String? ?? 'pending',
      plusOne: json['plusOne'] as bool? ?? false,
      dietaryRestrictions: json['dietaryRestrictions'] as String?,
      tableId: json['tableId'] as String?,
      side: json['side'] as String?,
      notes: json['notes'] as String?,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'tenantId': tenantId,
      'name': name,
      'email': email,
      'phone': phone,
      'rsvpStatus': rsvpStatus,
      'plusOne': plusOne,
      'dietaryRestrictions': dietaryRestrictions,
      'tableId': tableId,
      'side': side,
      'notes': notes,
    };
  }

  static List<Guest> fromJsonList(List<dynamic> jsonList) {
    return jsonList
        .map((json) => Guest.fromJson(json as Map<String, dynamic>))
        .toList();
  }
}

File: backend/src/main/java/com/weddingwire/guest/Guest.java
package com.weddingwire.guest;

import com.weddingwire.common.BaseEntity;
import jakarta.persistence.*;
import lombok.*;
import java.util.UUID;

@Entity
@Table(name = "guests")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Guest extends BaseEntity {

    @Column(name = "tenant_id", nullable = false)
    private UUID tenantId;

    @Column(nullable = false)
    private String name;

    private String email;

    private String phone;

    @Column(nullable = false)
    private String side = "mutual";

    @Column(name = "party_size", nullable = false)
    private Integer partySize = 1;

    @Column(name = "rsvp_status", nullable = false)
    private String rsvpStatus = "pending";

    private String dietary;

    @Column(name = "table_id")
    private UUID tableId;

    @Column(name = "save_the_date_sent", nullable = false)
    private Boolean saveTheDateSent = false;

    @Column(name = "invite_sent", nullable = false)
    private Boolean inviteSent = false;

    private String notes;
}
